tests/ttnn/unit_tests/test_global_semaphore.py

```diff
@@ -0,0 +1,34 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import torch
+import ttnn
+
+
+@pytest.mark.parametrize("enable_async_mode", (False, True), indirect=True)
+def test_global_semaphore(device, enable_async_mode):
+    tensix_cores0 = ttnn.CoreRangeSet(
+        {
+            ttnn.CoreRange(
+                ttnn.CoreCoord(0, 0),
+                ttnn.CoreCoord(3, 3),
+            ),
+        }
+    )
+    tensix_cores1 = ttnn.CoreRangeSet(
+        {
+            ttnn.CoreRange(
+                ttnn.CoreCoord(4, 4),
+                ttnn.CoreCoord(4, 4),
+            ),
+        }
+    )
+    global_sem0 = ttnn.create_global_semaphore(device, tensix_cores0, 1)
```

Comment:
Can it be device.create_***?

also, what is 1?

Reply:
1 is initial semaphore value the user wants it to be initialized to, could be any int.

As for `device.create_***`, I kept this similar to other apis we have, ex events. Those take device as an arg instead of being a member function of device.